src/main/java/com/fittracker/fittracker/rest/WeightRestController.java

```diff
@@ -0,0 +1,42 @@
+package com.fittracker.fittracker.rest;
+
+import com.fittracker.fittracker.entity.Weight;
+import com.fittracker.fittracker.rest.exception.WeightNotFoundException;
+import com.fittracker.fittracker.service.WeightService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+
+@RestController
+public class WeightRestController {
+
+    private WeightService weightService;
+
+    @Autowired
+    public WeightRestController(WeightService weightService) {
+        this.weightService = weightService;
+    }
+
+    @PostMapping("/weight")
+    @ResponseStatus(HttpStatus.CREATED)
+    public WeightResponse addWeight(@RequestBody @Valid Weight theWeight) {
+        theWeight.setId(0);
```

Comment:
As long as our `WeightRepository` extends `CrudRepository<Weight,Integer>` it does nothing. When there will appear need for custom repository setting id to 0 will result in forcing `EntityManager.merge(Entity entity)` to make insert instead of trying to update an non existing record in database.

Reply:
> As long as our `WeightRepository` extends `CrudRepository<Weight,Integer>` it does nothing.

So let's remove it.

> When there will appear need for custom repository...

Then we'll add it back.